### PR TITLE
New async APIs in DbTransaction

### PR DESCRIPTION
**Pull request by divega**

### CODE REVIEW

File: xml/System.Data.Common/DbTransaction.xml [hunk 1]
```diff
@@ -137,10 +137,19 @@
...
+        <param name="cancellationToken">A token to cancel the asynchronous operation.</param>
+        <summary>Asynchronously commits the database transaction.</summary>
+        <returns>A <see cref="P:System.Threading.Tasks.Task" /> representing the asynchronous operation.</returns>
+        <remarks>
+          <format type="text/markdown"><![CDATA[  
+  
+## Remarks  
+ The default implementation of this asynchronous method delegates to its synchronous counterpart and returns a completed `Task`, potentially blocking the calling thread.
+ 
+ Data providers that support [asynchronous programming](~/docs/framework/data/adonet/asynchronous-programming.md) should override the default inmplementation using asynchronous I/O operations.
```
Comment (divega) [inline xml/System.Data.Common/DbTransaction.xml]:
```suggestion
 Data providers that support [asynchronous programming](~/docs/framework/data/adonet/asynchronous-programming.md) should override the default implementation using asynchronous I/O operations.
```

File: xml/System.Data.Common/DbTransaction.xml [hunk 2]
```diff
@@ -367,9 +376,19 @@
...
-        <remarks>To be added.</remarks>
+        <summary>Asynchronously diposes the transaction object.</summary>
+        <returns>A <see cref="P:System.Threading.Tasks.ValueTask" /> representing the asynchronous operation.</returns>
+        <remarks>
+          <format type="text/markdown"><![CDATA[  
+  
+## Remarks  
+ The default implementation of this asynchronous method delegates to its synchronous counterpart and returns a completed `ValueTask`, potentially blocking the calling thread.
+ 
+ Data providers that support [asynchronous programming](~/docs/framework/data/adonet/asynchronous-programming.md) should override the default inmplementation using asynchronous I/O operations.
```
Comment (divega) [inline xml/System.Data.Common/DbTransaction.xml]:
```suggestion
 Data providers that support [asynchronous programming](~/docs/framework/data/adonet/asynchronous-programming.md) should override the default implementation using asynchronous I/O operations.
```

File: xml/System.Data.Common/DbTransaction.xml [hunk 3]
```diff
@@ -475,10 +494,19 @@
...
+        <param name="cancellationToken">A token to cancel the asynchronous operation.</param>
+        <summary>Asynchronously rolls back a transaction from a pending state.</summary>
+        <returns>A task representing the asynchronous operation.</returns>
+        <remarks>
+          <format type="text/markdown"><![CDATA[  
+  
+## Remarks  
+ The default implementation of this asynchronous method delegates to its synchronous counterpart and returns a completed `Task`, potentially blocking the calling thread.
+ 
+ Data providers that support [asynchronous programming](~/docs/framework/data/adonet/asynchronous-programming.md) should override the default inmplementation using asynchronous I/O operations.
```
Comment (divega) [inline xml/System.Data.Common/DbTransaction.xml]:
```suggestion
 Data providers that support [asynchronous programming](~/docs/framework/data/adonet/asynchronous-programming.md) should override the default implementation using asynchronous I/O operations.
```

File: xml/System.Data.Common/DbTransaction.xml [hunk 4]
```diff
@@ -36,7 +36,7 @@
     </Interface>
   </Interfaces>
   <Docs>
-    <summary>The base class for a transaction.</summary>
+    <summary>Defines the core behavior of database transactions and provides a base for derived classes.</summary>
```
Comment (divega) [inline xml/System.Data.Common/DbTransaction.xml]:
```suggestion
    <summary>Defines the core behavior of database transactions and provides a base for provider specific implementations.</summary>
```

**Comment by mairaw** [comment]
Good catch on those typos!

**Pull request merged**